test-snapshot-binaries/test-rpath-linux.py

```diff
@@ -1,25 +1,31 @@
 # Tests that DT_RPATH is correct for the dummy repl executable on Linux.
 # REQUIRES: platform=Linux
 # RUN: %{readelf} -d %{repl_swift} | %{FileCheck} %s
-# CHECK: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN/../lib/swift/linux
+# CHECK: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN/../lib/swift/linux/
 #
 # Tests that DT_RUNPATH is correct for the Swift stdlib and other libraries on Linux.
-# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/libswiftCore.so | %{FileCheck} --check-prefix CHECK-CORE %s
+# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/%{linux_arch}/libswiftCore.so | %{FileCheck} --check-prefix CHECK-CORE %s
+# CHECK-CORE: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN{{[^/]}}
 # CHECK-CORE-NOT: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}:/usr/lib/swift/linux
 #
 # RUN: %{readelf} -d %{package_path}/usr/lib/libsourcekitdInProc.so | %{FileCheck} --check-prefix CHECK-SK %s
+# CHECK-SK: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN/swift/linux/
 # CHECK-SK-NOT: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}:/usr/lib/swift/linux
 #
-# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/libswiftRemoteMirror.so | %{FileCheck} --check-prefix CHECK-RM %s
+# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/%{linux_arch}/libswiftRemoteMirror.so | %{FileCheck} --check-prefix CHECK-RM %s
+# CHECK-RM: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN{{[^/]}}
 # CHECK-RM-NOT: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}:/usr/lib/swift/linux
 #
-# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/libswift_Differentiation.so | %{FileCheck} --check-prefix CHECK-SD %s
+# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/%{linux_arch}/libswift_Differentiation.so | %{FileCheck} --check-prefix CHECK-SD %s
+# CHECK-SD: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN{{[^/]}}
 # CHECK-SD-NOT: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}:/usr/lib/swift/linux
 #
-# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/libswiftGlibc.so | %{FileCheck} --check-prefix CHECK-SG %s
+# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/%{linux_arch}/libswiftGlibc.so | %{FileCheck} --check-prefix CHECK-SG %s
+# CHECK-SG: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN{{[^/]}}
 # CHECK-SG-NOT: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}:/usr/lib/swift/linux
 #
-# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/libswiftSwiftOnoneSupport.so | %{FileCheck} --check-prefix CHECK-SON %s
+# RUN: %{readelf} -d %{package_path}/usr/lib/swift/linux/%{linux_arch}/libswiftSwiftOnoneSupport.so | %{FileCheck} --check-prefix CHECK-SON %s
+# CHECK-SON: {{.*}} {{\(RPATH\)|\(RUNPATH\)}} {{.*}}$ORIGIN{{[^/]}}
```

Comment:
As long as I'm updating this file, I added these `$ORIGIN` checks too, which are needed for the runtime libraries to find each other.